File: backend/api/schemas/vton.py
import uuid
from datetime import datetime
from enum import Enum

from pydantic import BaseModel


class ClothType(str, Enum):
    upper_body = "upper_body"
    lower_body = "lower_body"
    dress = "dress"


class JobStatus(str, Enum):
    queued = "queued"
    processing = "processing"
    completed = "completed"
    failed = "failed"


class VTONGenerateRequest(BaseModel):
    garment_photo_id: uuid.UUID
    model_photo_id: uuid.UUID
    cloth_type: ClothType


class VTONJobCreateResponse(BaseModel):
    job_id: uuid.UUID
    status: JobStatus
    created_at: datetime


class VTONJobStatusResponse(BaseModel):
    job_id: uuid.UUID
    status: JobStatus
    created_at: datetime
    started_at: datetime | None = None
    completed_at: datetime | None = None
    result_url: str | None = None
    error_reason: str | None = None
    retry_count: int = 0

File: frontend/lib/api/vton.ts
import { apiFetch } from "@/lib/api";

export interface VTONJobHistoryItem {
  job_id: string;
  status: "queued" | "processing" | "completed" | "failed";
  cloth_type: "upper_body" | "lower_body" | "dress";
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
  result_url: string | null;
  error_reason: string | null;
  retry_count: number;
}

export interface VTONJobHistoryResponse {
  items: VTONJobHistoryItem[];
  total: number;
  page: number;
  page_size: number;
}

export async function getVTONJobs(
  page = 1,
  pageSize = 50
): Promise<VTONJobHistoryResponse> {
  return apiFetch(`/api/vton/jobs?page=${page}&page_size=${pageSize}`);
}
